packages/firestore/src/local/indexeddb_schema_converter.ts

```diff
@@ -240,9 +249,9 @@ export class SchemaConverter implements SimpleDbSchemaConverter {
     }
 
     if (fromVersion < 14 && toVersion >= 14) {
-      p = p.next(() => {
-        createFieldIndex(db);
-      });
+      p = p
+        .next(() => createFieldIndex(db))
+        .next(() => this.runOverlayMigration(db, simpleDbTransaction));
```

Comment:
Same comment as in other PR - you should run this migration as version 14 and move indexing to 15.